corio.py

```diff
@@ -296,68 +299,68 @@ def main(options):
-                                                     options.number_of_nodes)
-
+        parsed_input[each] = yaml_parser.test_parser(each, options.number_of_nodes)
     for _, value in parsed_input.items():
         for test_key, test_value in value.items():
             LOGGER.info("Test Values : %s", value)
             if 'operation' in test_value.keys():
                 test_value['operation'] = function_mapping[
                     test_value['operation']]
                 value[test_key] = test_value
-
-    LOGGER.info("Parsed input files : ")
-    LOGGER.info(pformat(parsed_input))
-    processes = []
-    for test_plan, test_plan_value in parsed_input.items():
-        process = multiprocessing.Process(target=schedule_test_plan,
-                                          args=(test_plan, test_plan_value,
-                                                commons_params))
-        processes.append(process)
-    # TODO: Add support to schedule support bundle and health check periodically
+            if jira_flg:
+                if test_value["TEST_ID"] in tests_details:
+                    tests_to_execute[test_value["TEST_ID"]] = tests_details[test_value["TEST_ID"]]
+                    tests_to_execute[test_value["TEST_ID"]]['start_time'] = test_value['start_time']
+                    tests_to_execute[test_value["TEST_ID"]
+                                     ]['result_duration'] = test_value['result_duration']
     corio_start_time = datetime.now()
     LOGGER.info("Parsed input files : ")
     LOGGER.info(pformat(parsed_input))
+    LOGGER.info("Test to be executed.")
+    LOGGER.info(tests_to_execute)
     processes = {}
-
     for test_plan, test_plan_value in parsed_input.items():
         process = multiprocessing.Process(target=schedule_test_plan,
-                                          args=(test_plan, test_plan_value,
-                                                commons_params))
+                                          args=(test_plan, test_plan_value, commons_params))
         processes[test_plan] = process
-
     LOGGER.info(processes)
-
     # TODO: Add support to schedule support bundle and health check periodically
-    sched_job = schedule.every(30).minutes.do(log_status,
-                                              parsed_input=parsed_input,
-                                              corio_start_time=corio_start_time,
-                                              test_failed=None)
-
+    sched_job = schedule.every(30).minutes.do(
+        log_status, parsed_input=parsed_input, corio_start_time=corio_start_time, test_failed=None)
     try:
         for process in processes.values():
             process.start()
         terminate = False
         terminated_tp = None
+        test_ids = list()
         while True:
             time.sleep(1)
             schedule.run_pending()
+            if jira_flg:
+                jira_obj.update_jira_status(
```

Comment:
Update jira status should not be called after each second, 
this while loop is intended to poll the process status and hence jira update operation should not be part of this.

Reply:
JIRA api call will be called only when it is need as per time. Please refer update_jira_status.